Idp manager

nworr opened the pull request:
Add an admi module to enable/disable IDP plugins

Enabled IDP are still `/app/system/mainconfig.ini.php` , but disabled ones still need to be know

So I add`declareIDPPlugin` event 

ref #27

nworr committed:
feat : add idpadmin module

nworr committed:
feat(testapp) : add idpadmin to testapp

nworr committed:
feat(testapp) add command to declare 1st user

nworr committed:
feat IDP are now declared using events

laurentj commented on docs/en/events.md:
```diff
@@ -102,3 +102,9 @@ See the `Jelix\Authentication\LoginPass\AuthLPCanResetPasswordEvent` class.
 
 
 
+`declareIDPlugin`
```
je propose de plutôt nommer l'évènement `AuthAdminGetIDPPlugin`

laurentj commented on modules/authadmin/controllers/idpadmin.classic.php:
```diff
@@ -0,0 +1,109 @@
+<?php
```
je propose de nommer le module `authadmin` au lieu de idpadmin. Il portera d'autres pages de configuration générale pour l'authentification. Le contrôleur pourrait être nommé idpadmin plutôt que default (à terme default sera plutôt une sorte de page d'accueil de l'admin auth)

laurentj commented on modules/idpadmin/install/install.php:
```diff
@@ -0,0 +1,19 @@
+        jAcl2DbManager::createRightGroup($groupName, 'idpadmin~default.rights.group.name');
+
+        // Add right subject
+        jAcl2DbManager::createRight('idpadmin.view', 'idpadmin~default.idp.view', $groupName);
```
droits : `auth.idpadmin.*` plutôt que `idpadmin.*`

laurentj commented on modules/idpadmin/lib/AdminUiEventListener.php:
```diff
@@ -0,0 +1,31 @@
+            $uim = $event->uiManager;
+
+            $adminMenu = new SubMenu('admin', 'Administration', 10);
+            $adminMenu->addJelixLinkItem(jLocale::get('idpadmin~default.navigation.menu.idp'), 'idpadmin~default:index', array(), 'address-book');
```
intitulé de l'item de menu "authentification"

laurentj commented on modules/idpadmin/controllers/default.classic.php:
```diff
@@ -0,0 +1,109 @@
+
+    public function save()
+    {
+        $modif = new IniModifier(jApp::appPath('app/system/mainconfig.ini.php'));
```
la sauvegarde doit être faite dans var/config/liveconfig. Par contre pour la lecture, via jApp::config()

nworr commented:
rebased and updated
<img width="1253" height="487" alt="Screenshot 2026-02-09 at 15-16-52 Auth test app" src="https://github.com/user-attachments/assets/9eb4f189-48b8-4266-bbca-0e2d106d5b32" />
